File: Exercism Codes/Java/Hamming.java
public class Hamming {
    private String leftStrand;
    private String rightStrand;
    public Hamming(String leftStrand, String rightStrand) {
        if(leftStrand.isEmpty() && !rightStrand.isEmpty()) {
            throw new IllegalArgumentException("left strand must not be empty.");
        }
        if(rightStrand.isEmpty() && !leftStrand.isEmpty()) {
            throw new IllegalArgumentException("right strand must not be empty.");
        }
        if(leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }
        this.leftStrand = leftStrand;
        this.rightStrand = rightStrand;
    }

    public int getHammingDistance() throws IllegalArgumentException {
        char[] left = leftStrand.toCharArray();
        char[] right = rightStrand.toCharArray();
        int distance = 0;
        for(int i=0; i<left.length; i++){
            if(left[i] != right[i]){
                distance++;
            }
        }
        return distance;
    }
}

File: Exercism Codes/JavaScript/hamming.js
export const compute = (a, b) => {
    if(a.length != b.length)
        throw new Error('strands must be of equal length');
    else {
        let c = 0;
        for(let i = 0; i < a.length; i++) {
            if(a[i] != b[i]) c++;
        }
        return c;
    }
};
